pkg/skaffold/test/custom/custom.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2021 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package custom
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os/exec"
+	"runtime"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/list"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/color"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+)
+
+const Windows string = "windows"
+
+type Runner struct {
+	customTest     latest.CustomTest
+	testWorkingDir string
+}
+
+// New creates a new custom.Runner.
+func New(cfg docker.Config, wd string, ct latest.CustomTest) (*Runner, error) {
+	return &Runner{
+		customTest:     ct,
+		testWorkingDir: wd,
+	}, nil
+}
+
+// Test is the entrypoint for running custom tests
+func (ct *Runner) Test(ctx context.Context, out io.Writer, _ []build.Artifact) error {
+	if err := ct.runCustomCommand(ctx, out); err != nil {
+		return fmt.Errorf("running custom test command: %w", err)
+	}
+
+	return nil
+}
+
+func (ct *Runner) runCustomCommand(ctx context.Context, out io.Writer) error {
+	test := ct.customTest
+
+	// Expand command
+	command, err := util.ExpandEnvTemplate(test.Command, nil)
+	if err != nil {
+		return fmt.Errorf("unable to parse test command %q: %w", test.Command, err)
+	}
+
+	logrus.Infof("Running custom test command %q", command)
```

Comment:
We can get rid of this `logrus.Infof` as it not adding any extra information but repetition of what is below.